### PR TITLE
Issue 588: Renaming alterStream to updateStream to make the terminology consistent

### LINKED ISSUE #588: updateStream and alterStream being used in the Controller code inconsistently
**Problem description**
The controller code uses update and alter interchangeably to describe operations involving updating a stream configuration. We need to make this consistent, either use only alter or update everywhere.

**Problem location**
Controller APIs and code base.

**Suggestions for an improvement**
Use only alter or update and not both.

### PR DESCRIPTION
**Pull request by skrishnappa**
**Change log description**
Alter stream and update stream are being used interchangeably in pravega code. Here we update the code and documentation to consistently use 'update' since its a standard terminology for modify APIs.

**Purpose of the change**
Fixes #588 

**What the code does**
Replaces all occurrences of alter stream with update stream

**How to verify it**
No functionality change, all existing tests should pass.

### CODE REVIEW

File: client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java [hunk 1]
```diff
@@ -587,42 +592,49 @@ public void testCreateStream() throws Exception {
...
-        updateStreamStatus = controllerClient.alterStream(StreamConfiguration.builder()
+        updateStreamStatus = controllerClient.updateStream(StreamConfiguration.builder()
                                                                   .streamName("stream5")
                                                                   .scope("scope1")
                                                                   .scalingPolicy(ScalingPolicy.fixed(1))
                                                                   .build());
         AssertExtensions.assertThrows("Should throw Exception", updateStreamStatus, throwable -> true);
+
+        updateStreamStatus = controllerClient.updateStream(StreamConfiguration.builder()
+                .streamName("stream6")
```
Comment (fpj) [inline client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java]: Indentation here is different from the one used above.

Comment (skrishnappa) [inline client/src/test/java/io/pravega/client/stream/impl/ControllerImplTest.java]: Fixed.

**Comment by tkaitchuck** [comment]
If we are going to change its name on streams we should also change `alterConfig` on readerGroup to `updateConfig`

**Comment by tkaitchuck** [comment]
Also, perhaps it should be `updateStreamConfig` rather than `updateStream`.

**Comment by skrishnappa** [comment]
>> Also, perhaps it should be updateStreamConfig rather than updateStream.

@tkaitchuck not so sure about this one, since this seems consistent with the other APIs - createStream, deleteStream,  etc. I'll update this if I get a +1 from others too - @fpj @bayarsaikhan ?

**Comment by fpj** [comment]
@skrishnappa @tkaitchuck I can see where the comment about `updateStreamConfig` is coming from and it is a pretty reasonable suggestion. Here is how I'm thinking about it.

The operations in `StreamManager` are more about the stream metadata than the data itself. As such, `updateStream` unambiguously refers to updating the stream metadata and the specific way os via config parameters. Adding `Config` to the name makes it explicit, but I don't feel it causes any confusion between an update to the stream data and an update to the stream metadata give the class the method is in. If we update it, then we all break the symmetry with the other methods.

If my reasoning isn't broken and any of you feels strongly about it, then I'd rather leave as is, but please speak up if you feel it is a necessary change.

**Comment by tkaitchuck** [comment]
updateStream is fine but we should still change ReaderGroup to be similar.

**Comment by skrishnappa** [comment]
>> updateStream is fine but we should still change ReaderGroup to be similar.

@tkaitchuck , thats done, ReaderGroup#alterConfig is modified to ReaderGroup#updateConfig

**Pull request merged**